misc/userscripts/qutebrowser_viewsource

```diff
@@ -24,9 +24,15 @@
 # Caveat: Does not use authentication of any kind. Add it in if you want it to.
 #
 
+TERMINAL=${TERMINAL:-urxvt}
+EDITOR=${EDITOR:-vim}
+
 path=/tmp/qutebrowser_$(mktemp XXXXXXXX).html
+curl "$QUTE_URL" > "$path"
 
-curl "$QUTE_URL" > $path
-urxvt -e vim "$path"
+unset _command
+[[ $EDITOR =~ emacs|gedit|leafpad|atom ]] && _command="$EDITOR $path"
+[ -z "$_command" ] && _command="$TERMINAL -e \"vim $path\""
 
+eval "$_command"
```

Comment:
The `-e` flag is not standard, so various terminals handle it differently. [rxvt](http://linux.die.net/man/1/rxvt) and [xterm](http://linux.die.net/man/1/xterm) have a special `-e` flag, which causes anything following it to be parsed as arguments of the command to be executed rather than the terminal itself. On the other hand some terminals, which don't implement their own command line parsers (probably all VTE based terminals), have an `--execute=COMMAND` option with an `-e` shortcut, which is much simpler to parse (e.g. they can use existing libraries very easily).

If you want to make exceptions, I think it is the rxvt/xterm way that is in minority. But you'd have to evaluate dozens of terminals individually to make the right decision for all of them.